### PR TITLE
feat: upgradability using truffle plugin

### LINKED ISSUE #134: Upgradeability to Storage Contract
Make `StorageManager` contract Upgradeable, and allow to deploy it via proxy using OpenZeppelin.

### PR DESCRIPTION
**Pull request by julianmrodri**
* Prepares the `storageManager` contract for Upgradability. Uses the OpenZeppelin Upgrades Truffle Plugin.
* Adapted migrations and tests (required to ensure we always use `account[0]` for deployment of the proxies).

Closes #134

### CODE REVIEW

File: test/Staking.test.js [hunk 1]
```diff
@@ -5,21 +5,23 @@ const {
...
+const upgrades = require('@openzeppelin/truffle-upgrades')
 const { toBN, asciiToHex, padRight } = require('web3-utils')
 const expect = require('chai').expect
 
 const Staking = artifacts.require('Staking')
 const StorageManager = artifacts.require('StorageManager')
 const ERC20 = artifacts.require('MockERC20')
 
-contract('Staking', ([staker, randomPerson]) => {
+contract('Staking', ([randomPerson, staker]) => {
```
Comment (nduchak) [inline test/Staking.test.js:16]: why do you revert this?

**Review by nduchak — APPROVED** [review]
LGTM

File: test/Staking.test.js (hunk 1 above)
Comment (julianmrodri) [reply test/Staking.test.js:16]: The plugins have still an open issue that is that you cannot specify the `from` address. It uses the `accounts[0]` from the accounts that Truffle provides. So I needed the first account to be the one Deploying the proxies. The easiest way was to revert the order to achieve that and have the one that deploys first.

Comment (nduchak) [reply test/Staking.test.js:16]: Ahh. Ok, make sense now). Thanks

Comment (julianmrodri) [reply test/Staking.test.js:16]: https://github.com/OpenZeppelin/openzeppelin-upgrades/issues/85

**Pull request merged**